dashboard/src/components/AppView/AppView.tsx

```diff
@@ -166,33 +165,40 @@ export default function AppView() {
   }, [cluster, dispatch, namespace, releaseName, pluginObj]);
 
   useEffect(() => {
-    if (!app || !app.apiResourceRefs) {
-      return () => {};
-    }
-
-    // If there are at least some resource types (ingresses, deployments) that are populated
-    // then avoid re-requesting the refs.
-    if (Object.values(appViewResourceRefs).some(ref => ref.length)) {
+    if (!app?.apiResourceRefs) {
       return () => {};
     }
 
     const parsedRefs = parseResources(app.apiResourceRefs);
     setAppViewResourceRefs(parsedRefs);
     return () => {};
-  }, [app, cluster, kinds, appViewResourceRefs]);
+  }, [app?.apiResourceRefs]);
```

Comment:
Just cleaned up the dependencies (ie. we don't want to fire this effect whenever the app changes, only when the resource refs change).